### DISCUSSION TITLE
Lua conditionals do not evaluate for every time point

### DISCUSSION THREAD

**Discussion by vikgus1** [facontidavide/PlotJuggler#625] [Q&A] [answered]
Hi,

Maybe I'm misunderstanding something about how Lua works, but it seems impossible for me to get the built in Lua interpreter to evaluate conditionals for each point in the time series. For example, the script
```
if(value > 0) then
   return 1
else
   return 0
end
```
Seems to return whichever conditional was evaluated first, for all points in the time series. E.g. even if my input contains both positive and negative values, if the first value is positive, then each point in the returned time series will be 1, and vice versa.
It seems that Lua interpreter evaluates the conditionals for the first point in the time series, and then applies that flow to all subsequent points. Is there a way around this?
I'm using version 3.3.2 on Ubuntu 20.04.
Huge appreciation for any reply!

**Comment by facontidavide** [comment, marked as answer]
that is very weird, I am 100% sure that it should not happen.

Give me a way to reproduce the issue, step by step with the LATEST version of PlotJuggler, and I will look into it

**Reply by vikgus1** [reply]
My apologies, I messed up the conditions. Thank you for the reply.
